### ISSUE TITLE
Additional head to the current architecture 

### ISSUE THREAD

**Issue by abhigoku10** [bubbliiiing/yolov4-pytorch#245]
@bubbliiiing hi thanks for sharing the code base I have a few queries on the idea which i want to implement 

Idea1: to add an additional classification head with a bounding box that performs gender classification for an only person  as shown below
![image](https://user-images.githubusercontent.com/22094053/146105754-58106e55-34b2-4ca4-b2dd-dd88e034f5e9.png)

Idea2: to add multilabel classifier head along with the current classification to perform multiattribute classification ie single bounding box and multi-label prediction as shown below 
![image](https://user-images.githubusercontent.com/22094053/146105921-c9ea2114-e4dd-4f7f-9255-5bcb65f03314.png)

can you share your thoughts on which one is more feasible to implement with the current code base  Thanks in advance

**Comment by bubbliiiing** [comment]
If it is necessary to modify the original code, I would suggest adding a prediction head to only distinguish what you want. The channel is 2 * 3, but the data loading part, data set production part and loss function need to be modified.
If it is an engineering implementation, you can use a classification network

**Comment by abhigoku10** [comment]
@bubbliiiing thanks for the response , so the above applies for both idea1 and idea2 ha ?
1. can modify the dataset loading part to take up additional box and classes 
2. for loss we can use the same classification loss currently used or bce loss 
3. adding the prediction head  and modifying the anchors is where i am having doubt on how to proceed 
4. and in validation of the model should we consider the additional head also is one more doubt

**Comment by bubbliiiing** [comment]
Theoretically, both idea1 and idea2 can be implemented. Idea1 needs to be modified a lot, but idea2 needs to be modified very little

Idea1 does not need to modify the anchor box, and additional category information needs to be updated in 2007_ train.txt. Maybe the classification header does not need to be modified. The original classification header is used directly, because sigmoid is used to calculate the probability, which can be classified in essence. The tag handling method needs to be modified a little during loss construction

The decoding process directly outputs multiple categories
